src/Chapter1.hs

```diff
@@ -554,7 +557,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z 
+    | (x < y && x > z) || (x < z && x > y) = x
+    | (y < x && y > z) || (y < z && y > x) = y
+    | (z < y && z > x) || (z < x && y > z) = z
+    | otherwise = if x == y then x else z
```

Comment:
You can use `>=` and `<=` instead of `<` and `>`. And with this approach, you'll be able to replace last two guards with a single `otherwise`